core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -76,10 +76,25 @@ import org.apache.spark.util.random.{BernoulliSampler, PoissonSampler, Bernoulli
  * on RDD internals.
  */
 abstract class RDD[T: ClassTag](
-    @transient private var sc: SparkContext,
+    @transient private var _sc: SparkContext,
     @transient private var deps: Seq[Dependency[_]]
   ) extends Serializable with Logging {
 
+  if (classOf[RDD[_]].isAssignableFrom(elementClassTag.runtimeClass)) {
```

Comment:
Similarly, this should perhaps be a warning instead of an exception in order to avoid any possibility of breaking odd corner-case 1.2.1 apps.  I'll change this to a warning and leave the `sc` getter as an exception.